camel-prod-maven-plugin/src/main/java/org/l2x6/cq/camel/maven/prod/CamelProdExcludesMojo.java

```diff
@@ -310,9 +310,23 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                 additionalFiles);
 
         /* Remove non-prod components from camel-allcomponents in the copy */
-        new PomTransformer(originalFullTreeCopyDir.resolve("core/camel-allcomponents/pom.xml"), charset,
-                simpleElementWhitespace)
-                        .transform(Transformation.removeDependency(true, true, gavtcs -> !includes.contains(gavtcs.toGa())));
+        /* Only do this if camel-allcomponents exists */
+        Path allComponents = originalFullTreeCopyDir.resolve("core/camel-allcomponents/pom.xml");
+        if (allComponents.toFile().exists()) {
+            new PomTransformer(allComponents, charset,
+                    simpleElementWhitespace)
+                            .transform(
+                                    Transformation.removeDependency(true, true, gavtcs -> !includes.contains(gavtcs.toGa())));
+        }
+
+        /* Test to see if catalog/camel-allcomponents/pom.xml exists, possibly use that path instead */
+        allComponents = originalFullTreeCopyDir.resolve("catalog/camel-allcomponents/pom.xml");
+        if (allComponents.toFile().exists()) {
+            new PomTransformer(allComponents, charset,
+                    simpleElementWhitespace)
+                            .transform(
+                                    Transformation.removeDependency(true, true, gavtcs -> !includes.contains(gavtcs.toGa())));
+        }
```

Comment:
The code in those two if blocks, is the same, is it not? So we basically need to perform it for the first existing allcomponents dir - is the following maybe more DRY?

```suggestion
        /* Find the fist existing allcomponents dir */
        Path allComponents = Stream.of(
                "core/camel-allcomponents/pom.xml",
                "catalog/camel-allcomponents/pom.xml" // since Camel 3.17 or so
        )
                .map(originalFullTreeCopyDir::resolve)
                .filter(Files::isRegularFile)
                .findFirst()
                .orElseThrow(() -> new IllegalStateException(
                        "Could not find allcomponents module; searched in core/camel-allcomponents/pom.xml and catalog/camel-allcomponents/pom.xml"));
        /* Remove non-prod components from camel-allcomponents in the copy */
        new PomTransformer(allComponents, charset,
                simpleElementWhitespace)
                        .transform(
                                Transformation.removeDependency(true, true, gavtcs -> !includes.contains(gavtcs.toGa())));

```

Reply:
Applied suggested changes